### PR TITLE
Don't ignore figsize in df.boxplot

### PR DESCRIPTION
**Pull request by huguesv**
Work in progress from PyCon sprints

Fix for https://github.com/pandas-dev/pandas/issues/11959

It really doesn't feel like a proper fix, but I'm not sure what a proper fix would be (without changing matplotlib).  The matplotlib `gca()` ends up calling `figure()`, but doesn't let us pass in a `figsize` to `gca()`.

Any advice? FYI this works, ie it fixes the behavior for the code snippet in the bug report. It would obviously break in multithreaded scenarios, but I'm not sure if that's a consideration or not.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Thanks @huguesv. I think the issue is somewhere around https://github.com/pandas-dev/pandas/blob/49ec31bbaeca81a6f58fc1be26fe80f3ac188cdd/pandas/plotting/_core.py#L2009

This simplest fix would be to wrap that call in a `plt.rc_context` like

```python

In [8]: with plt.rc_context({'figure.figsize': figsize}):
   ...:     ax = plt.gca()
 ```

Alternatively you could modify `_gca` to take an `rc` (a dictionary) like `plt.rc_context`. And just pass that through. Make sense?

**Comment by TomAugspurger** [comment]
Sorry I got confused by the diff. I see that you found the right line :)

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -1871,14 +1880,8 @@ def plot_series(data, kind='line', ax=None,                    # Series unique
...
     import matplotlib.pyplot as plt
-    """
-    If no axes is specified, check whether there are existing figures
-    If there is no existing figures, _gca() will
-    create a figure with the default figsize, causing the figsize=parameter to
-    be ignored.
-    """
     if ax is None and len(plt.get_fignums()) > 0:
-        ax = _gca()
+        ax = _gca(figsize)
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: I would maybe revert this change, since we pass figsize through to `_plot`, presumably it's handled there.

**Comment by TomAugspurger** [comment]
You can add a release note in `doc/source/whatsnew/v0.20.2.txt` under the "Other Enhancements" section, about how `DataFrame.boxplot` now respects the `figsize` keyword for non-grouped boxplots.

File: pandas/plotting/_core.py (hunk 1 above)
Comment (huguesv) [inline pandas/plotting/_core.py]: I missed that, thanks. Changing it back.

**Comment by huguesv** [comment]
Oops, changed the wrong whatsnew file, let me fix that.

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -49,9 +49,13 @@ def _get_standard_kind(kind):
     return {'density': 'kde'}.get(kind, kind)
 
 
-def _gca():
+def _gca(figsize=None):
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py]: Can you change `_gca` to take an optionally dictionary? So like

```python
def _gca(rc=None):
    with plt.rc_context(rc):
        return plt.gca()
```

that way, we aren't limited to just `figsize` when we call gca. You'd call it like `ax = _gca({'figure.figsize': figsize})`

**Comment by TomAugspurger** [comment]
Did you have a test earlier? I thought you did, but maybe not. Anyway, could you add one in `pandas/tests/plotting/test_boxplot_method.py`  Just need to verify that `ax = df.boxplot(figsize=(4, 4))` or whatver actually has the correct figsize.

File: pandas/plotting/_core.py (hunk 2 above)
Comment (huguesv) [inline pandas/plotting/_core.py]: I can do that. Note that I can't unconditionally overwrite figure.figsize, because if it's not specified, it's None, and that would overwrite the default in rcParams. That's why I have it pass in an empty dict when it's None.  Anyway, I can push that one level up the call stack like you mention.

Comment (TomAugspurger) [inline pandas/plotting/_core.py]: Ah that's unfortunate. Still probably best to handle it outside.

**Comment by huguesv** [comment]
Added test.

File: pandas/tests/plotting/test_boxplot_method.py [hunk 3]
```diff
@@ -160,6 +160,13 @@ def test_boxplot_empty_column(self):
         df.loc[:, 0] = np.nan
         _check_plot_works(df.boxplot, return_type='axes')
 
+    @slow
+    def test_figsize(self):
+        df = DataFrame(np.random.rand(10, 5), columns=['A', 'B', 'C', 'D', 'E'])
+        result = df.boxplot(return_type='axes', figsize=(12,8))
```
Comment (TomAugspurger) [inline pandas/tests/plotting/test_boxplot_method.py]: You'll need a comma between the 12 and 8, otherwise our style cheker will complain.

File: pandas/tests/plotting/test_boxplot_method.py [hunk 4]
```diff
@@ -160,6 +160,13 @@ def test_boxplot_empty_column(self):
         df.loc[:, 0] = np.nan
         _check_plot_works(df.boxplot, return_type='axes')
 
+    @slow
+    def test_figsize(self):
+        df = DataFrame(np.random.rand(10, 5), columns=['A', 'B', 'C', 'D', 'E'])
```
Comment (TomAugspurger) [inline pandas/tests/plotting/test_boxplot_method.py]: this line might be a bit longn too. `flake8 pandas/tests/plotting/test_boxplot_method.py` should tell you.

**Review by TomAugspurger — APPROVED** [review]
Looks great. Let me know if you notice that all the tests pass before I do.

**Comment by TomAugspurger** [comment]
The one test failure seemed unrealted. I've restarted it to be safe though: https://travis-ci.org/pandas-dev/pandas/jobs/235323266

**Comment by huguesv** [comment]
All tests passed! :)

**Comment by TomAugspurger** [comment]
@huguesv thanks. Nice job!
